test/testcafe/spec/uiDemo_spec.js

```diff
@@ -0,0 +1,28 @@
+import { RequestMock } from 'testcafe';
+import UiDemoPageObject from '../framework/page-objects/UiDemoPageObject';
+import uiDemoResponse from '../../../playground/mocks/data/idp/idx/_ui-demo.json';
+
+import compareScreenshot from '../../../vrtUtil/vrtUtil';
+
+const uiDemoMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(uiDemoResponse);
+
+async function setup(t) {
+  const pageObject = new UiDemoPageObject(t);
+  await pageObject.navigateToPage();
+  await t.expect(pageObject.formExists()).eql(true);
+  return pageObject;
+}
+
+fixture('UI demo').meta('gen3', false).meta('gen2', false);
+
+test
+  .requestHooks(uiDemoMock)('UI demo VRT', async t => {
+    const pageObject = await setup(t);
+
+    // freeze the spinner element so screenshots are consistent
+    await pageObject.stopSpinnerAnimation();
+
+    await compareScreenshot(t, { fullPage: true, strictMode: true });
```

Comment:
instead of having multiple screenshots we take one of the fullpage.  This way it is easier to manage and all the ui components are shown together as a whole, without being cut off